### PR TITLE
countl_zero improvment for sse2 and avx2 + simplification of bit_ceil bit_floor

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: benchmarks/module/core/bit_ceil/regular/bit_ceil.hpp [hunk 1]
```diff
@@ -13,12 +13,9 @@ int main()
   auto lmin = eve::valmin(eve::as<EVE_VALUE>());
   auto lmax = eve::valmax(eve::as<EVE_VALUE>());
 
-  auto const std__bit_ceil = [](EVE_VALUE x) { return x > 0 ? x : EVE_VALUE(-x); };
```
Comment (jfalcou) [inline benchmarks/module/core/bit_ceil/regular/bit_ceil.hpp:16]: why not using std::bit_ceil when possible ?

File: benchmarks/module/core/bit_floor/regular/bit_floor.hpp [hunk 2]
```diff
@@ -7,18 +7,28 @@
...
+       return std::bit_floor(x);
+    else return x;
+  };
 
   eve::bench::experiment xp;
-  run<EVE_VALUE> (EVE_NAME(std__bit_floor) , xp, std__bit_floor, arg0);
+  using v_t = EVE_VALUE;
+  if constexpr(std::is_integral_v<v_t>)
+  {
+    std::cout << std::is_integral_v<EVE_VALUE> << std::endl;
```
Comment (jfalcou) [inline benchmarks/module/core/bit_floor/regular/bit_floor.hpp:28]: this needs cleanup

File: include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp [hunk 3]
```diff
@@ -9,7 +9,14 @@
...
 #include <eve/concept/value.hpp>
 #include <eve/detail/implementation.hpp>
+#include <eve/module/core/regular/bit_shr.hpp>
+#include <eve/module/core/regular/bit_notand.hpp>
+#include <eve/module/core/regular/convert.hpp>
+#include <eve/module/core/regular/sub.hpp>
+#include <eve/module/core/regular/min.hpp>
+#include <eve/module/core/regular/bit_cast.hpp>
 
+#include <iostream>
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp:19]: spurrious iostream

File: include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp [hunk 4]
```diff
@@ -23,13 +30,48 @@ namespace eve::detail
...
     constexpr auto c = categorize<r_t>();
 
-    if constexpr( current_api < avx512 || sizeof(T) < 4 ) return countl_zero.behavior(cpu_{}, opts, a0);
-    else if constexpr( c == category::uint64x8 ) return r_t(_mm512_lzcnt_epi64(a0));
-    else if constexpr( c == category::uint32x16) return r_t(_mm512_lzcnt_epi32(a0));
-    else if constexpr( c == category::uint64x4 ) return r_t(_mm256_lzcnt_epi64(a0));
-    else if constexpr( c == category::uint32x8 ) return r_t(_mm256_lzcnt_epi32(a0));
-    else if constexpr( c == category::uint64x2 ) return r_t(_mm_lzcnt_epi64(a0));
-    else if constexpr( c == category::uint32x4 ) return r_t(_mm_lzcnt_epi32(a0));
+//     if constexpr( sizeof(T) < 4)
```
Comment (jfalcou) [inline include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp:33]: remove commented code

**Review by jfalcou — CHANGES_REQUESTED** [review]

File: include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp:33]: done

File: include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/core/regular/impl/simd/x86/countl_zero.hpp:19]: done

File: benchmarks/module/core/bit_floor/regular/bit_floor.hpp (hunk 2 above)
Comment (jtlap) [reply benchmarks/module/core/bit_floor/regular/bit_floor.hpp:28]: done

File: benchmarks/module/core/bit_ceil/regular/bit_ceil.hpp (hunk 1 above)
Comment (jtlap) [reply benchmarks/module/core/bit_ceil/regular/bit_ceil.hpp:16]: done

**Review by jfalcou — APPROVED** [review]

**Pull request merged**